src/gpu/amd/sycl_hip_memory_storage_helper.hpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+* 
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_AMD_SYCL_HIP_MEMORY_STORAGE_HELPER_HPP
+#define GPU_AMD_SYCL_HIP_MEMORY_STORAGE_HELPER_HPP
+
+#include <optional>
+#include "gpu/amd/sycl_hip_scoped_context.hpp"
+#include "sycl/sycl_memory_storage.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace amd {
+
+#define CTX_IN_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::read>(&CTX_IN_STORAGE(arg), cgh)
+
+#define CTX_OUT_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::write>(&CTX_OUT_STORAGE(arg), cgh)
+
+#define CTX_SCRATCH_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::read_write>( \
+            ctx.get_scratchpad_grantor().get_memory_storage(arg).get(), cgh)
+
+template <::sycl::access_mode mode>
+class sycl_memory_arg_t {
```

Comment:
This file almost completely duplicates `sycl_cuda_memory_storage_helper.hpp`. I would suggest you to rename it to 
`sycl_memory_storage_helper.hpp` and put it in `src/sycl`.

Also, I think interface of `get_native_pointer` can be simplified by identifying backend at compile time.
```cpp

template <::sycl::access_mode mode>
class sycl_memory_arg_t {
#ifdef DNNL_SYCL_CUDA
    static constexpr auto be = ::sycl::backend::ext_oneapi_cuda
#endif

#ifdef DNNL_SYCL_HIP
    static constexpr auto be = ::sycl::backend::ext_oneapi_hip
#endif
/* . . . */
    template <typename T = void>
    T *get_native_pointer(const compat::interop_handle &ih) const { /* . . . */ }
/* . . . */
};
```
NOTE: Don't forget to update the following NVIDIA files:
```
./src/gpu/nvidia/cudnn_binary.cpp:19:#include "gpu/nvidia/sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_convolution.cpp:22:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_deconvolution.cpp:22:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_eltwise.cpp:22:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_inner_product.cpp:24:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_lrn.cpp:23:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_matmul_executor.hpp:24:#include "gpu/nvidia/sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_pooling.cpp:27:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_reorder.cpp:22:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_resampling.cpp:23:#include "sycl_cuda_memory_storage_helper.hpp"
./src/gpu/nvidia/cudnn_softmax.cpp:22:#include "sycl_cuda_memory_storage_helper.hpp"
```

Reply:
As per your suggestion we renamed to **sycl_memory_storage_helper.hpp** and placed it in src/sycl. 
And also updated  all the necessary files.